src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -1414,6 +1419,33 @@ static MixMonoToStereoF GetMonoToStereoMixer(const WAVEFORMATEXTENSIBLE *fmtext,
     return NULL;
 }
 
+// ------------------------------------------------------------------------------------------
+static HRESULT ReallocateMonoMixerBuffer(PaWasapiSubStream *subStream, UINT32 frames)
+{
+    assert(subStream->monoMixer != NULL);
+
+    // If not provided then use max buffer size of WASAPI device
+    if (frames == 0)
+        frames = subStream->bufferSize;
+
+    assert(frames != 0);
+
+    // Expand buffer if necessary
+    UINT32 monoBufferSize = frames * subStream->wavexu.ext.Format.nBlockAlign;
+    if (monoBufferSize > subStream->monoBufferSize)
+    {
+        subStream->monoBufferSize = monoBufferSize;
```

Comment:
Better to perform this assignment only once the reallocation succeeds

Reply:
No problem to adjust this implementation. Additionally I set `monoBufferSize` to 0 when memory allocation fails that looks logical because memory buffer gets freed in this case.